services/command-router-base/src/main/java/org/eclipse/hono/commandrouter/impl/CommandRouterServiceImpl.java

```diff
@@ -65,7 +69,7 @@
     private final DeviceConnectionInfo deviceConnectionInfo;
     private final MessagingClient<CommandConsumerFactory> commandConsumerFactories;
     private final Tracer tracer;
-    private final Deque<String> tenantsToEnable = new LinkedList<>();
+    private final Deque<Pair<String, Integer>> tenantsToEnable = new LinkedList<>();
```

Comment:
How about clearing `tenantsToEnable` and cancelling a potentially existing timer in the `stop()` method?